client/src/main/java/org/kakara/client/game/world/io/GroupedChunkIO.java

```diff
@@ -42,12 +43,12 @@ public void run() {
                     }
```

Comment:
Ensure that this threads ends peacefully before the termination of the program. This could be the source of the corruption.